### PR TITLE
Typos & basic grammar errors fixing

### PR DESCRIPTION
**Pull request by Fourchaux**
This PR is a "rewriting" of ocaml#817

To resume:
Inside comments of ml {i} files, only indisputable (uhuh...) typos
as well as some basic grammar errors are corrected.

Directories checked :
asmcomp - bytecomp - debugger - lex - ocamldoc - parsing - stdlib - typing - utils.

Certainly, typos are still present.
Please report.

### CODE REVIEW

**Comment by gasche** [comment]
The typos fixed seem reasonable to me, thanks!

Do we know why the testsuite fails on Windows? (Is this a line-ending issue?)

It would be nice of you to add a Changes entry in either the "Manual and documentation" or the "Internal/compiler-libs changes", crediting your work and also the people that provided reviews and feedback.

**Comment by dra27** [comment]
AppVeyor fail is my fault, and unrelated (test should be restarting shortly).

This all looks good to me, thanks - it's ready for @damiendoligez to merge after 4.05.0 is released.

File: typing/ctype.ml [hunk 1]
```diff
@@ -1315,7 +1315,7 @@ let subst env level priv abbrev ty params args body =
     raise exn
 
 (*
-   Only the shape of the type matters, not whether is is generic or
+   Only the shape of the type matters, not whether is generic or
```
Comment (murmour) [inline typing/ctype.ml]: ... not whether _it's_ generic or not.

File: stdlib/camlinternalFormat.ml [hunk 2]
```diff
@@ -2055,15 +2055,15 @@ let fmt_ebb_of_string ?legacy_behavior str =
     invalid_format_message str_ind
       "non-zero widths are unsupported for %c conversions"
   in
-  (* Raise Failure with a friendly error message about an option dependencie
+  (* Raise a Failure with a friendly error message about an option dependency
```
Comment (murmour) [inline stdlib/camlinternalFormat.ml]: I wouldn't use an article before "Failure".

**Review by murmour — COMMENTED** [review]
Looks good. I've added some minor remarks.

File: stdlib/camlinternalFormat.ml (hunk 2 above)
Comment (damiendoligez) [reply stdlib/camlinternalFormat.ml]: Me neither, and I would put `Failure` in brackets (ocamldoc-style).

Comment (Fourchaux) [reply stdlib/camlinternalFormat.ml]: OK. @dra27 also said :
Either Raise [Failure] or "Raise a Failure exception".

Raise [Failure] is the winner :)

Thank you.

File: typing/ctype.ml (hunk 1 above)
Comment (Fourchaux) [reply typing/ctype.ml]: Fixed. TY.

File: ocamldoc/odoc_print.ml [hunk 3]
```diff
@@ -47,7 +47,7 @@ exception Use_code of string
 (** Return the given module type where methods and vals have been removed
    from the signatures. Used when we don't want to print a too long module type.
    @param code when the code is given, we raise the [Use_code] exception is we
-   encouter a signature, to that the calling function can use the code rather
+   encounter a signature, to that the calling function can use the code rather
```
Comment (yallop) [inline ocamldoc/odoc_print.ml]: "is we encouter" ↝ "if we encouter"

"to that" ↝ "so that"

Comment (Fourchaux) [reply ocamldoc/odoc_print.ml]: Fixed. Thank you.

**Comment by mshinwell** [comment]
@Fourchaux please rebase.  It's high time we got this merged.

**Comment by mshinwell** [comment]
Looks like something's gone wrong with that rebase?

**Comment by Fourchaux** [comment]
Not sure what was wrong (incorrect git push I think) !

My 'github.com:Fourchaux/ocaml.git' is good now (I hope ...)


But this PR  displays '...merge ... commits into ocaml:trunk from unknown repository'.
How to solve that? Should I open a new PR?

**Comment by gasche** [comment]
Yes, feel free to open a new PR.

**Pull request closed without merging**